### ISSUE TITLE
Crashes on saving

### ISSUE THREAD

**Issue by PatDyn** [monocasual/giada#641]
**Environment**
- OS: Manjaro
- Giada version: 0.24.0

**Describe the bug**
When saving a project, giada crashes quite frequently.

**To Reproduce**

1. Create a project. 
3. Add some some sample channels
4. Add some samples to the channels
5. Try to save
6. May work with some samples
7. Will crash on other samples on each try  

---

1. Create a project
2. Remove some columns
3. Add some midi channels
4. Add another column
5. Try to save  

It seems, that this sometimes also happens when editing   
MIDI input. But I can not reproduce this consistently.  

**Expected behavior**
I would like to be able to save my projects without fear of losing the last half hour of my work.

**Comment by gvnnz** [comment]
Hey @PatDyn , thanks for reporting. Is there a project you can share where the issue is easily reproducible?

**Comment by PatDyn** [comment]
Gladly! 
This project almost surely crashes on saving, when adding samples to any of the channels. 
[SynthSet-Reduced14.gprj.tar.gz](https://github.com/monocasual/giada/files/11646358/SynthSet-Reduced14.gprj.tar.gz)

I actually ran giada on the console, trying to catch some meaningful messages. But nothing turned up so far. 
I'll try again though.

**Comment by PatDyn** [comment]
This is the log for loading a sample and trying to save to the same project. What's not reflected in the logs is a dialog appearing at the end of the saving process, saying "Unable to save project".

The behavior is the same for v0.24 and v0.25

`[init] Giada 0.25.0
[init] Build date: Jun  3 2023
[init] Release build
[init] Dependencies:
[init]   FLTK - 1.4.0
[init]   RtAudio - 6.0.0beta1
[init]   RtMidi - 5.0.0
[init]   Libsamplerate
[init]   Libsndfile - libsndfile-1.0.31
[init]   JSON for modern C++ - 3.10.5
[init]   JUCE - 7.0.5
[KA] Compiled RtAudio APIs: 4
  JACK
  PulseAudio
  ALSA
  Dummy
[KM] Compiled RtMidi APIs: 1
  ALSA
[StorageApi::loadProject] Load project from /home/erik/Documents/Musik/LiveTechSessions/2023-05-23-AlphaSetup/SynthSet-Reduced14.gprj
[mixer::disable] disabled
[mixer::reset] buffers ready - maxFramesInLoop=529200, framesInBuffer=1024
[mixer::enable] enabled
[waveManager::create] new Wave created, 145986 frames
[StorageApi::storeProject] Project dir created: /home/erik/Documents/Musik/LiveTechSessions/2023-05-23-AlphaSetup/SynthSet-Reduced14.gprj/.gprj

`

**Comment by gvnnz** [comment]
Thanks for the reduced project @PatDyn . I couldn't reproduce the issue, but I can tell from the log there's something wrong with the project directory:

```
[StorageApi::storeProject] Project dir created: /home/user/Documents/Musik/LiveTechSessions/2023-05-23-AlphaSetup/SynthSet-Reduced14.gprj/.gprj
```

There's an extra `/.gprj` at the end of the file path that shouldn't be there. Can you please show me a screenshot of the file browser when you're about to save the project? Thanks!

**Comment by PatDyn** [comment]
Hi, this is the screenshot of the project dir:
![dirBeforeSaving](https://github.com/monocasual/giada/assets/37243484/37d498b6-8a88-47b3-80b7-5d899bfed914)

**Comment by gvnnz** [comment]
Thanks, the folder content is messed up indeed. What name are you giving to the project? Does it contain dots, backslashes or any weird symbols?

**Comment by PatDyn** [comment]
The name is "SynthSet-Reduced16". Also the Path to the folder: "/home/user/Documents/Musik/LiveTechSessions/2023-05-23-BetaSetup/" does not contain any special charactes except dashes.

**Comment by gvnnz** [comment]
Hey @PatDyn, this bug is absolutely not reproducible on my machine (Debian 12 with XFCE latest version). Could you please record a quick video where I can see the exact steps you take?

**Comment by PatDyn** [comment]
Ok, I recorded a <1min vid, which hopefully gives you more insight :)
https://github.com/monocasual/giada/assets/37243484/756ae314-40a0-4db1-aa60-19bc8805bdf6

I used giada v0.24.0 for that.

**Comment by gvnnz** [comment]
Thanks @PatDyn! I think I see the problem now: there's an extra `/` at the end of the project name. So instead of `some-project.gprj/` it should be `some-project.gprj`. This is a bug in how Giada filters the file name: do you still get the error if you remove the `/` manually?

**Comment by PatDyn** [comment]
Alright, i did some checking. Did some clicks, loaded some samples and then saved a few versions. 
I could not consistently create a situation, where the `/` appeared at the end of the filename. 
But manually adding and then saving lead to the initially described behavior. Thanks for the hint!

**Comment by gvnnz** [comment]
Great. The issue has been already fixed in c800a829c4c7f040fb7e93e31842c6939194b0bb (and later commits) and will be included in the upcoming v0.25.1 release. Thank you!
